src/items/static-items.md

```diff
@@ -8,16 +8,20 @@
 > [^extern-safety]: The `safe` and `unsafe` function qualifiers are only
 >   allowed semantically within `extern` blocks.
 
-A *static item* is similar to a [constant], except that it represents a precise
-memory location in the program. All references to the static refer to the same
-memory location. Static items have the `static` lifetime, which outlives all
-other lifetimes in a Rust program. Static items do not call [`drop`] at the
-end of the program.
+A *static item* is similar to a [constant], except that it represents an allocated object in the
+program that is initialized with the initializer expression. All references and raw pointers to the
+static refer to the same allocated object. Static items have the `static` lifetime, which outlives
+all other lifetimes in a Rust program. Static items do not call [`drop`] at the end of the program.
+
+This allocated object is disjoint from all other `static` objects as well as heap-allocated and
+stack-allocated variables. However, the storage of immutable `static` items can overlap with objects
+that do not themselves have a unique address, such as promoteds and `const` items.
```

Comment:
this seems basically good to me, thank you!

though I wonder if https://github.com/rust-lang/reference/blob/23ce619966541bf2c80d45fdfeecf3393e360a13/src/destructors.md?plain=1#L295 should be linked from [promoteds]?